wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -573,6 +586,49 @@ private Optional<ExpressionFactory<String>> getExpressionFactory(RelationalTranf
     return ctx.getEngine().getExpressionFactory(StringExpressionFactoryType.SQL);
   }
 
+  /**
+   * This method emits all metrics for the given list of directives
+   *
+   * @param directives a list of Wrangler directives
+   * @param metrics CDAP {@link Metrics} object using which metrics can be emitted
+   * @throws DirectiveLoadException
+   */
+  private void emitDirectiveMetrics(List<Directive> directives, Metrics metrics) throws DirectiveLoadException {
+    for (Directive directive : directives) {
+      List<EntityMetricDef> entityMetricDefs = new ArrayList<>();
+
+      // Emit directive usage metric only if directive is present in system directive registry
+      if (registry.get(Contexts.SYSTEM, directive.define().getDirectiveName()) != null) {
+        entityMetricDefs.add(getDirectiveUsageMetric(directive.define().getDirectiveName()));
+      }
+
+      if (directive instanceof EntityMetrics) {
+        entityMetricDefs.addAll(((EntityMetrics) directive).getMetrics());
+      }
+
+      for (EntityMetricDef entityMetricDef : entityMetricDefs) {
+        Metrics child = metrics.child(entityMetricDef.getTags());
+        switch (entityMetricDef.getType()) {
+          case COUNTER:
+            child.countLong(entityMetricDef.getName(), entityMetricDef.getValue());
+            break;
+          case GAUGE:
```

Comment:
I can't really think of a situation where a directive would want to emit a gauge. Do we have a use case? If not it's better not to have code for it.